src/renderers/shared/fiber/ReactFiberContext.js

```diff
@@ -129,13 +131,15 @@ exports.pushContextProvider = function(workInProgress : Fiber, didPerformWork :
     instance.__reactInternalMemoizedMergedChildContext = mergedContext;
   }
 
-  index++;
-  contextStack[index] = mergedContext;
-  didPerformWorkStack[index] = didPerformWork;
+  push(contextStackCursor, mergedContext, workInProgress);
+  push(didPerformWorkStackCursor, didPerformWork, workInProgress);
 };
 
 exports.resetContext = function() : void {
-  index = -1;
+  reset();
```

Comment:
Seems slightly weird to bypass the contexts and talk directly to the stack they use, no?

But sure. I can change that.